### ISSUE TITLE
FR: default to bolder fonts to mitigate low color contrast

### ISSUE THREAD

**Issue by ortoslon** [crawl/crawl#3116]
Several foreground colors from https://github.com/crawl/crawl/blob/4198be8af4a5659e449fb70e20ff6c1d3c4c5eb3/crawl-ref/source/tilefont.cc do not contrast enough against black background for "normal text":

BLUE https://webaim.org/resources/contrastchecker/?fcolor=0052FF&bcolor=000000
BROWN https://webaim.org/resources/contrastchecker/?fcolor=A55B00&bcolor=000000
DARKGREY https://webaim.org/resources/contrastchecker/?fcolor=525252&bcolor=000000

One way to mitigate that is to use bolder fonts: see the first note at https://www.w3.org/TR/WCAG21/#dfn-large-scale

Luckily, the bundled https://github.com/crawl/crawl-fonts/blob/6d3046974f35189a679a4d84b254efe82d1653b3/DejaVuSansMono.ttf has a sharp jump in boldness between 17 ppem and 18 ppem (because of strong Truetype hinting):

https://www.motaitalic.com/tools/font-tester/latin/
![image](https://user-images.githubusercontent.com/79928123/235908644-d8498556-ec8c-4172-a5fa-dcde1700309a.png)

That means it may be sufficient to raise default Mono sizes in local tiles to 18.

Side effect: Youtube and Twitch videos of DCSS will become more legible because bolder fonts survive lossy compression better.

A comparison:

With current default sizes (empty init.txt) on Win10 on a 1920*1080 screen:
![image](https://user-images.githubusercontent.com/79928123/235372506-15c2fd01-1a63-44ba-a051-920d23c5328f.png)

With tile_font_crt_size = 18, tile_font_msg_size = 18, tile_font_stat_size = 18:
![image](https://user-images.githubusercontent.com/79928123/235372573-abfc45fa-b832-4717-88b2-7a5ecd5e6be1.png)

**Comment by PleasingFungus** [comment]
Done. This feels very weird, but maybe I'll get used to it.

**Comment by jv84** [comment]
I didn't have any contrast issues with 1920*1080 screen on Linux.
Now the font size is too big and I have to tweak options.

**Comment by rawlins** [comment]
I'm a bit late to the party on this but I think on the whole it's probably better to have defaults calibrated to smaller screens rather than larger (in absence of something more dynamic).

Also, in the screenshots in the original FR, something is wrong with the font rendering (it has the hallmarks of over-aggressive sub-pixel rendering) and I don't think making the fonts bigger was the correct solution for exactly what's going on there.

**Comment by rawlins** [comment]
For reference, here's how the old defaults look with correct font rendering on a 4K screen. I think it's fine, I'm sort of inclined to revert?

<img width="1599" alt="Screenshot 2023-07-19 at 9 03 31 AM" src="https://github.com/crawl/crawl/assets/6290753/ed5b6900-9849-4160-b5df-91140113b662">

**Comment by ortoslon** [comment]
@rawlins, the font size on your screenshots is much larger than 17 ppem (physical pixels, that is) because of the 4K screen.

The only option that I remember affecting font rendering without changing font sizes is tile_font_ft_light, which is false by default: https://github.com/crawl/crawl/blob/eda263fcf18bcd25282576ecdee048bca832059c/crawl-ref/docs/options_guide.txt#L2389. I will attach screenshots with "true" later.

**Comment by rawlins** [comment]
> the font size on your screenshot is much larger than 17 ppem (physical pixels, that is) because of the 4K screen.

That is true, but it doesn't change the fact that in the original screenshots, the rendering is off (and hopefully, not representative -- though it seems conceivable that there is a windows-specific issue). In fact, the over-aggressive font blurring is visible in the window title in those screenshots..

**Comment by PleasingFungus** [comment]
I've reverted this per request.

For posterity, latent bugs which this triggered:
- A large empty space appeared at the bottom of the skill menu. (Goratrix)
- The sidebar became wider than needed. (Goratrix)
- Labels when mousing over inventory items in the bottom-right area overlapped with the inventory itself. (MIC32)

These should be filed separately and fixed eventually to support folks who do want to increase font size, but I don't have time to fix them at present, unfortunately. The UI code is complex enough that fixes didn't appear trivial.

**Comment by ortoslon** [comment]
The promised screenshot on Win10 on a 1920*1080 screen with init.txt consisting only of
tile_font_ft_light = true
![image](https://github.com/crawl/crawl/assets/79928123/99d0064c-4389-4be1-ab64-2d5fa09d1f85)

It is indeed slightly bolder at the cost of more blurriness. In other words, color contrast at the cost of edge contrast.

@rawlins, the white-on-teal window title is probably blurrier just because of colors: the subpixel rasterizer is limited in which subpixels it can use. The full explanation is in http://rastertragedy.com/ but it's quite a rabbithole. Either way, I doubt the window title rasterization shares any settings with contents of the window because all Crawl's text on our screenshots so far is wholepixel-antialiased (I've checked the absence of multicolored fringes by zooming).

**Comment by ortoslon** [comment]
For more posterity, the gist of the trouble with fonts on "legacy" screens such as 1920*1080 LCDs:

In "plain" sans-serif fonts of reasonable size for reading, horizontal strokes will be either 1 px wide, 2 px wide, or blurry: subpixel rendering doesn't help with horizontals when subpixels are laid out the usual way. The bundled Dejavu Mono jumps in boldness between 17 and 18 ppem exactly to avoid the blurry valley. The discontinuous jumps in boldness are made possible with Truetype hinting: embedded programs that nudge character outlines to edges of the pixel grid. I guess tile_font_ft_light=true makes the Freetype rasterizer skip executing some of instructions in the hinting programs but I'm not sure.

Vertical strokes should be as thick as horizontals, preferably thicker because of optical illusions. (Fonts that break this rule even slightly look "off": https://www.fontshop.com/families/ff-balance). But at least they can have fractional thickness and not look too blurry, but only if subpixel rendering is enabled.

In both of my original screenshots (17 and 18 ppem with default non-light hinting) verticals are as thick as horizontals. That gives the impression that horizontals are thicker because of the optical illusion. The rasterizer cannot make verticals only slightly thicker without either whole-pixel blurriness or subpixel rendering.
